### PR TITLE
Prevents registering for sections with deleted teachers.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Fixes https://app.honeybadger.io/projects/3240/faults/30788900.

### CODE REVIEW

File: dashboard/test/controllers/followers_controller_test.rb [hunk 1]
```diff
@@ -94,6 +94,21 @@ class FollowersControllerTest < ActionController::TestCase
     assert_equal @chris_section, follower.section
   end
 
+  test "student_user_new errors when joining a section with deleted teacher" do
+    @laurel.update(deleted_at: Time.now)
```
Comment (aoby) [inline dashboard/test/controllers/followers_controller_test.rb:98]: This variable name makes me sad :(

Comment (aoby) [reply dashboard/test/controllers/followers_controller_test.rb:98]: no need to change it. I'm just saying

Comment (aoby) [reply dashboard/test/controllers/followers_controller_test.rb:98]: Not to get too off the rails here in this PR, but her birthday is tomorrow, so it's especially on my mind

Comment (ashercodeorg) [reply dashboard/test/controllers/followers_controller_test.rb:98]: The variable name saddened me too. And though I thought about changing, I decided leaving it was a good way to remember Laurel.

Though I wasn't close to her (like you and others), nor did I know her for that long, Laurel had a surprising impact on me.

**Comment by ashercodeorg** [comment]
Nope. But I'm fairly certain this is the cause, based on the example IDs in HoneyBadger.

**Pull request merged**